lib/src/migrator.dart

```diff
@@ -276,6 +303,13 @@ class _Migrator extends RecursiveStatementVisitor implements ExpressionVisitor {
     }
   }
 
+  /// Emits a warning [message] with [span] and its context.
+  void _warn(String message, FileSpan span) {
+    print(message);
+    print(span.highlight());
```

Comment:
You probably want to use [`SourceSpan.message`](https://pub.dartlang.org/documentation/source_span/latest/source_span/SourceSpan/message.html) here; it's built to associate messages with spans.